### PR TITLE
Port to MacOS

### PR DESCRIPTION
**Pull request by jwhonce**
* Refactor Tunnel to support selecting port for remote sshd
* Refactor ssh tunnel to support MacOS version of ssh
* Refactor Tunnel.close() to find and kill off zombie siblings
* Add psutil dependency
* Add logging setup, letting library produce debugging records
* Clean up Tunnel API
* Fix test_runner.sh to propagate returncode to caller

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@jwhonce all kinds of test unhappiness

File: contrib/python/pypodman/pypodman/lib/config.py [hunk 1]
```diff
@@ -125,8 +157,8 @@ def resolve_configuration(self, args):
             if dir_ is None:
                 continue
             with suppress(OSError):
-                with open(os.path.join(dir_, 'pypodman/pypodman.conf'),
-                          'r') as stream:
+                with open(os.path.join(dir_,
+                                       'pypodman/pypodman.conf')) as stream:
```
Comment (TomSweeneyRedHat) [inline contrib/python/pypodman/pypodman/lib/config.py]: Just double checking to verify that you meant to remove the 'r' here....

Comment (jwhonce) [reply contrib/python/pypodman/pypodman/lib/config.py]: @TomSweeneyRedHat Yup read is default.

**Comment by mheon** [comment]
@jwhonce Looking at the failing tests more.

It looks like a call to `inotify_init1` is failing with `EMFILE`. This could be max open files for the process, but it could also be maximum inotify instances for a user being reached. The limit (/proc/sys/fs/inotify/max_user_instances) is per-UID, and set to 128 on my system. Might be more likely than exhausting the default 1024 file descriptors?

**Comment by jwhonce** [comment]
@mheon Below are the crazy numbers I tried last night just for debugging. (I reordered the output so it made more sense.)  I didn't tweak fs.inotify.max_user_instances but I will and let you know the results.
 
+ ulimit -n 999999
+ ulimit -n
999999
+ sysctl fs.inotify.max_user_watches=999999
fs.inotify.max_user_watches = 999999
+ setenforce 0
setenforce: SELinux is disabled
+ prlimit --nofile=999999:999999 podman --storage-driver=vfs --root=/tmp/podman.c3ilOF6z0B/crio --runroot=/tmp/podman.c3ilOF6z0B/crio-run --cni-config-dir=/tmp/podman.c3ilOF6z0B/cni/net.d varlink --timeout=0 unix:/tmp/podman.c3ilOF6z0B/podman/io.projectatomic.podman
+ prlimit --pid 8864 --nofile
RESOURCE DESCRIPTION                SOFT   HARD UNITS
NOFILE   max number of open files 999999 999999 files
+ lsof
+ wc -l
787
+ sysctl fs.file-nr
fs.file-nr = 1024	0	788581
+ sysctl fs.file-max
fs.file-max = 788581

**Comment by mheon** [comment]
Tests are green, @rhatdan @baude @TomSweeneyRedHat PTAL

**Comment by jwhonce** [comment]
@lsm5 FYI -- There are dependency updates in the spec files.

File: contrib/spec/podman.spec.in [hunk 2]
```diff
@@ -203,6 +203,7 @@ BuildRequires: python3-varlink
 Requires: python3-setuptools
 Requires: python3-varlink
 Requires: python3-dateutil
+Requires: python3-psutil
```
Comment (lsm5) [inline contrib/spec/podman.spec.in]: sgtm, just curious, is this something needed because of the MacOS porting?

Comment (jwhonce) [reply contrib/spec/podman.spec.in]: @lsm5 Yes, I needed to get more aggressive with killing off child processes.  Process groups have different defaults. I will also need this package for the Windows port.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**